scrapy/pqueues.py

```diff
@@ -26,17 +26,8 @@ def _path_safe(text):
 
 
 class ScrapyPriorityQueue:
-    """A priority queue implemented using multiple internal queues (typically,
-    FIFO queues). It uses one internal queue for each priority value. The internal
-    queue must implement the following methods:
-
-        * push(obj)
-        * pop()
-        * close()
-        * __len__()
-
-    Optionally, the queue could provide a ``peek`` method, that should return the
```

Comment:
Queue classes got a new optional method recently, see https://github.com/scrapy/scrapy/pull/5112; I think it should be mentioned - not here, but in the queue interface docs.

Reply:
https://github.com/scrapy/scrapy/pull/5112 doesn't provide a motivation(intended usage) for method introduction. Without motivation It is hard to come up with proper protocol. Hey @elacuesta can you provide more information here? Most important part is what should we do in case of transient errors during peek?

Reply:
My personal motivation to add this (first in https://github.com/scrapy/queuelib/pull/38) was solving a specific problem I'm having in [scrapy-playwright](https://github.com/scrapy-plugins/scrapy-playwright), which is not really relevant for scrapy itself, at least at the moment.
The method is not currently used by any internal scrapy component, and I don't expect it to be used in the near future, given how long scrapy has managed without it. In any case, I think it's a pretty standard part of any queue interface, and it's just there in case any custom code could benefit from it (even more so after documenting the scheduler at #3559). Additionally, some scrapy tests inherit from the queuelib test suite and were failing because of the absence of `peek` (fixed in the mentioned PR, #5112).
Regarding `TransientError`, if I understand correctly that's something that could be raised within `push`, I don't think we need to consider it for `peek`, which should probably mimic whatever `pop` does (return `None` as documented in this PR).